test/Integration.t.sol

```diff
@@ -0,0 +1,224 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.21;
+
+import "dss-test/DssTest.sol";
+
+import { Domain } from "dss-test/domains/Domain.sol";
+import { OptimismDomain } from "dss-test/domains/OptimismDomain.sol";
+import { TokenBridgeDeploy } from "deploy/TokenBridgeDeploy.sol";
+import { L2TokenBridgeSpell } from "deploy/L2TokenBridgeSpell.sol";
+import { L1TokenBridgeInstance } from "deploy/L1TokenBridgeInstance.sol";
+import { L2TokenBridgeInstance } from "deploy/L2TokenBridgeInstance.sol";
+import { TokenBridgeInit, BridgesConfig } from "deploy/TokenBridgeInit.sol";
+import { L1TokenBridge } from "src/L1TokenBridge.sol";
+import { L2TokenBridge } from "src/L2TokenBridge.sol";
+import { GemMock } from "test/mocks/GemMock.sol";
+
+interface SuperChainConfigLike {
+    function guardian() external returns (address);
+    function paused() external view returns (bool);
+    function pause(string memory) external;
+}
+
+interface L1CrossDomainMessengerLike {
+    function superchainConfig() external returns (address);
+    function paused() external view returns (bool);
+}
+
+contract IntegrationTest is DssTest {
+
+    Domain l1Domain;
+    OptimismDomain l2Domain;
+
+    // L1-side
+    DssInstance dss;
+    address PAUSE_PROXY;
+    address L1_MESSENGER;
+    address l1GovRelay;
+    address escrow;
+    L1TokenBridge l1Bridge;
+    GemMock l1Token;
+
+    // L2-side
+    address l2GovRelay;
+    GemMock l2Token;
+    L2TokenBridge l2Bridge;
+    address L2_MESSENGER;
+
+    constructor() {
+        vm.setEnv("FOUNDRY_ROOT_CHAINID", "1"); // used by ScriptTools to determine config path
+        // Note: need to set the domains here instead of in setUp() to make sure their storages are actually persistent
+        string memory config = ScriptTools.loadConfig("config");
+        l1Domain = new Domain(config, getChain("mainnet"));
+        l2Domain = new OptimismDomain(config, getChain("base"), l1Domain);
+    }
+
+    function setUp() public {
+        l1Domain.selectFork();
+        l1Domain.loadDssFromChainlog();
+        dss = l1Domain.dss();
+        PAUSE_PROXY = dss.chainlog.getAddress("MCD_PAUSE_PROXY");
+        vm.label(address(PAUSE_PROXY), "PAUSE_PROXY");
+
+        L1_MESSENGER = l2Domain.readConfigAddress("l1Messenger");
+        L2_MESSENGER = l2Domain.readConfigAddress("l2Messenger");
+        vm.label(L1_MESSENGER, "L1_MESSENGER");
+        vm.label(L2_MESSENGER, "L2_MESSENGER");
+
+        address l1GovRelay_ = vm.computeCreateAddress(address(this), vm.getNonce(address(this)) + 3); // foundry increments a global nonce across domains
```

Comment:
I'm a bit confused here, the `deployL2` has 3 contract deployments + switching the owner of the bridge, which should make an action as we are indeed moving the owner from the test address to the L2 govRelay. How is it that the nonce needs to increment 3 instead of 4?

Reply:
Already discussed.